src/IO/S3/getObjectInfo.cpp

```diff
@@ -46,93 +41,25 @@ namespace
         return client.HeadObject(req);
     }
 
-    Aws::S3::Model::GetObjectAttributesOutcome getObjectAttributes(const Aws::S3::S3Client & client, const String & bucket, const String & key, const String & version_id, bool for_disk_s3)
-    {
-        ProfileEvents::increment(ProfileEvents::S3GetObjectAttributes);
-        if (for_disk_s3)
-            ProfileEvents::increment(ProfileEvents::DiskS3GetObjectAttributes);
-
-        Aws::S3::Model::GetObjectAttributesRequest req;
-        req.SetBucket(bucket);
-        req.SetKey(key);
-
-        if (!version_id.empty())
-            req.SetVersionId(version_id);
-
-        req.SetObjectAttributes({Aws::S3::Model::ObjectAttributes::ObjectSize});
-
-        return client.GetObjectAttributes(req);
-    }
-
-    Aws::S3::Model::GetObjectOutcome getObjectDummy(const Aws::S3::S3Client & client, const String & bucket, const String & key, const String & version_id, bool for_disk_s3)
-    {
-        ProfileEvents::increment(ProfileEvents::S3GetObject);
-        if (for_disk_s3)
-            ProfileEvents::increment(ProfileEvents::DiskS3GetObject);
-
-        Aws::S3::Model::GetObjectRequest req;
-        req.SetBucket(bucket);
-        req.SetKey(key);
-
-        if (!version_id.empty())
-            req.SetVersionId(version_id);
-
-        /// Only the first byte will be read.
-        /// We don't need that first byte but the range should be set otherwise the entire object will be read.
-        req.SetRange("bytes=0-0");
-
-        return client.GetObject(req);
-    }
-
-
     /// Performs a request to get the size and last modification time of an object.
     /// The function performs either HeadObject or GetObjectAttributes request depending on the endpoint.
     std::pair<std::optional<ObjectInfo>, Aws::S3::S3Error> tryGetObjectInfo(
-        const Aws::S3::S3Client & client, const String & bucket, const String & key, const String & version_id,
-        const S3Settings::RequestSettings & request_settings, bool with_metadata, bool for_disk_s3)
+        const S3::Client & client, const String & bucket, const String & key, const String & version_id,
+        const S3Settings::RequestSettings & /*request_settings*/, bool with_metadata, bool for_disk_s3)
     {
-        if (request_settings.allow_head_object_request)
```

Comment:
If we don't use `GetObjectAttributes` requests anymore let's remove the setting `allow_head_object_request` too.